src/seq.rs

```diff
@@ -601,23 +654,41 @@ impl fmt::Debug for Event {
     }
 }
 
-/// Low level methods to set/get data on an Event. Don't use these directly, use generic methods on Event instead.
-pub trait EventData {
+/// Low level method to check if event type supports a given data type. Don't use directly, use generic methods on Event instead.
+pub trait HasEventData {
```

Comment:
I don't think that we need `take_ext` ... at least I can't think of any usecase: If you're creating the Event yourself, you usually don't need to get the data out again (because you have put it there in the first place), and if the Event is obtained from an ALSA call, it won't have a `Vec` stored inside.